doc/transceiver-monitor-hld.md

```diff
@@ -0,0 +1,151 @@
+
+
+### 1.2 Access eeprom from platform container ###
+
+Transceiver information eeprom can be accessed via read files(e.g. `/sys/bus/i2c/devices/2-0048/hwmon/hwmon4/qsfp9_eeprom`), different vendors may have these files under different folders, these folder need to be mounted to platform container so Xcvrd can access them. 
+
+
+For the convenience of implementation and reduce the time consuming, need to do enhancement to the `SfpUtilBase` class:
+
+1. `SfpUtilBase` internally should add the ability to read the eeprom and only pick up the interested bytes by given offset and number of bytes.
+
+2. `SfpUtilBase` will provide APIs `get_eeprom_sfp_info_dict(self, port_num)` and `get_eeprom_dom_info_dict(self, port_num)` to return `eeprom_if_dict` and `eeprom_dom_dict` separately, the interested values of these two dict are defined  in section 1.1.1 and 1.1.2.  In these two APIs can pick up these values from eeprom by provide the corresponding offset and number of bytes. 
+
+
+### 1.3 Transceiver plug in/out event ###
+
+Xcvrd need to be triggered by transceiver plug in/out event to refresh the transceiver info table.
+
+How to get this event is various on different platform, there is no common implementation available. 
+
+Here we define a common platform API to wait for this event in class `SfpUtilBase`: 
+
+    @abc.abstractmethod
+    def get_transceiver_change_event(self):
+        """
+        :returns: Boolean, True if call successful, False if not; 
+        dict for pysical port number and the SFP status. like {'0': 'PLUGIN', '31':'PLUGOUT'}
+        """
+        return 
+
+Each vendor need to implement this function in `SfpUtil` plugin.
+
+Xcvrd will call this API to wait for the sfp plug in/out event, following example code showing how this API will be called:
+
+    while True:
+        status, port_dict = platform_sfputil.get_transceiver_change_event()
+        if(status):
+            for key, value in port_dict.iteritems():
+                print("SFP on port: %s" was %s" % (key, value))
+                 
+
+
+### 1.4 Xcvrd daemon flow ###
+
+Xcvrd will spawn a thread to wait for the SFP plug in/out event, when event received, it will update the DB entries accordingly.
+
+A timer will be started to periodically refresh the DOM sensor information . 
+
+Detailed flow as showed in below chart: 
+
+![](https://github.com/keboliu/SONiC/blob/xcvrd-hld/images/transceiver_monitoring_hld/xcvrd_flow.svg)
+
+## 2. SNMP Agent Change ##
+
+### 2.1 MIB tables extension ###
+
+MIB table entPhysicalTable from [Entity MIB(RFC2737)](https://tools.ietf.org/html/rfc2737) need to be extended to support new OIDs.
+
+| OID | SNMP counter | Where to get the info in Sonic. | Example: |
+| --- | --- | --- | --- |
+| 1.3.6.1.2.1.47.1.1.1 | entPhysicalTable |   |   |
+| 1.3.6.1.2.1.47.1.1.1.1 | entPhysicalEntry |   |   |
+| 1.3.6.1.2.1.47.1.1.1.1.2. ifindex | entPhysicalDescr | Show interfaces alias | Xcvr for Ethernet29 |

```

Comment:
The ifindex should be replaced with index, which is Ifindex * 1000, as changed in requirement